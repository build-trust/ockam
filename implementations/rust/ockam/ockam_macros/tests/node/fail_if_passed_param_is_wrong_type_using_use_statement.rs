// This test checks that #[ockam_macros::node] causes a compile time error
// if the function is passed a param that is not of type `ockam::Context`
// The param is not fully qualified (ie. using `use` statement).

use std::string::String;

#[ockam_macros::node]
async fn main(ctx: String) -> Result<()> {
    Ok(())
}
