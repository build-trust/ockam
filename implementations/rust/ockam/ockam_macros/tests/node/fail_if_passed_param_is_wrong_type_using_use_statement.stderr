error: Expected `ockam::Context` found `String`
 --> tests/node/fail_if_passed_param_is_wrong_type_using_use_statement.rs:8:20
  |
8 | async fn main(ctx: String) -> Result<()> {
  |                    ^^^^^^

warning: unused import: `std::string::String`
 --> tests/node/fail_if_passed_param_is_wrong_type_using_use_statement.rs:5:5
  |
5 | use std::string::String;
  |     ^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/node/fail_if_passed_param_is_wrong_type_using_use_statement.rs:5:1
   |
5  | / use std::string::String;
6  | |
7  | | #[ockam_macros::node]
8  | | async fn main(ctx: String) -> Result<()> {
9  | |     Ok(())
10 | | }
   | |_^ consider adding a `main` function to `$DIR/tests/node/fail_if_passed_param_is_wrong_type_using_use_statement.rs`
